pkg/ipcache/metadata.go

```diff
@@ -158,12 +158,28 @@ func (m *metadata) setInjectedRevision(rev uint64) {
 // supplied revision. We may skip revisions, as the desired revision is bumped
 // every time prefixes are dequeued, but injection may fail. Thus, any revision
 // greater or equal to the desired revision is acceptable.
-func (m *metadata) waitForRevision(rev uint64) {
+func (m *metadata) waitForRevision(ctx context.Context, rev uint64) error {
+	// Allow callers to bail out by cancelling the context
+	cleanupCancellation := context.AfterFunc(ctx, func() {
+		// We need to acquire injectedRevisionCond.L here to be sure that the
+		// Broadcast won't occur before the call to Wait, which would result
+		// in a missed signal.
+		m.injectedRevisionCond.L.Lock()
+		m.injectedRevisionCond.Broadcast()
+		m.injectedRevisionCond.L.Unlock()
```

Comment:
Would it make sense to defer the `Unlock` here?

Reply:
With these short functions, it's a matter of personal preference I would argue. I can add a defer!

Reply:
Follow your style, I was just wondering if it wasn't safe!